packages/core/src/utils.ts

```diff
@@ -560,12 +560,14 @@ export function deepIncludes(value: any, pattern: any): boolean {
   return value === pattern;
 }
 
-function deepIncludesArray(array1: any[], array2: any[]): boolean {
-  return array1.every((value1) => array2.some((value2) => deepIncludes(value1, value2)));
+function deepIncludesArray(value: any[], pattern: any[]): boolean {
+  return pattern.every((patternVal) => value.some((valueVal) => deepIncludes(valueVal, patternVal)));
```

Comment:
It can be difficult to notice with both the logic fix and the argument name change, but previously every value of what is now called `value` was  being checked to exist in `pattern` instead of the other way around.  Also, the order of the two parameters passed to `deepIncludes` was inconsistent (aka swapped); effectively switching which object was being consider the pattern.